presto-hive/src/main/java/com/facebook/presto/hive/security/ranger/RangerBasedAccessControl.java

```diff
@@ -0,0 +1,516 @@
+            throw new RuntimeException(format("Request to %s failed: %s [Error: %s]", endPtUri, users, users.getResponseBody()));
+        }
+
+        return users.getValue();
+    }
+
+    @TestOnly
+    public void setRangerAuthorizer(RangerAuthorizer rangerAuthorizer)
+    {
+        this.rangerAuthorizer = rangerAuthorizer;
+    }
+
+    @TestOnly
+    public void setUsers(Users users)
+    {
+        this.users = users;
+    }
+
+    private static <T> T jsonParse(Response response, Class<T> clazz)
+            throws IOException
+    {
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(response.body().byteStream()));
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        return mapper.readValue(bufferedReader, clazz);
+    }
+
+    private boolean userExists(String userName)
+    {
+        return users.getvXUsers().stream()
+                .filter(user -> userName.equalsIgnoreCase(user.getName()))
+                .findAny().isPresent();
+    }
+
+    private Set<String> getGroupsForUser(String username)
+    {
+        if (userExists(username)) {
+            return users.getvXUsers().stream()
+                    .filter(user -> username.equalsIgnoreCase(user.getName()))
+                    .findFirst()
+                    .get()
+                    .getGroupNameList()
+                    .stream().collect(Collectors.toSet());
+        }
+        return null;
+    }
+
+    enum HiveAccessType
+    {
+        NONE, CREATE, ALTER, DROP, INDEX, LOCK, SELECT, UPDATE, USE, ALL, ADMIN
+    };
+
+    private boolean checkAccess(ConnectorIdentity identity, SchemaTableName tableName, String column, HiveAccessType accessType)
+    {
+        return rangerAuthorizer.authorizeHiveResource(tableName.getSchemaName(), tableName.getTableName(), column,
+                accessType.toString(), identity.getUser(), getGroupsForUser(identity.getUser()));
+    }
+
+    /**
+     * Check if identity is allowed to create the specified schema in this catalog.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    public void checkCanCreateSchema(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, String schemaName)
+    {
+    }
+
+    /**
+     * Check if identity is allowed to drop the specified schema in this catalog.
+     *
+     * @throws AccessDeniedException if not allowed
+     */
+    public void checkCanDropSchema(ConnectorTransactionHandle transactionHandle, ConnectorIdentity identity, AccessControlContext context, String schemaName)
+    {
+        SchemaTableName schemaTableName = new SchemaTableName(schemaName, schemaName);
+        if (!checkAccess(identity, schemaTableName, null, HiveAccessType.DROP)) {
+            denyCreateSchema(schemaTableName.toString(), format("Access denied - User [ $s ] does not have is not have [DROP] " +
```

Comment:
for the schema name argument - you should just say `schemaName` instead of `schemeTableName.toString()`

Reply:
ok